docs/backtracking/n-queens.mdx

```diff
@@ -0,0 +1,260 @@
+
+```
+*---*---*---*---*
+| X |   |   |   |
+*---*---*---*---*
+|   |   |   | X |
+*---*---*---*---*
+|   | X |   |   |
+*---*---*---*---*
+|   |   |   |   |
+*---*---*---*---*
+```
+
+Now the 4th queen cannot be placed in the 4th row as there exists no safe cell such that no two queens can attack each other. So we will return false and backtrack to the previous row and change the queen's position. We will repeat the same procedure untill we find the positions for the queens.
+Therefore, the answer for N=4 is:
+
+```
+*---*---*---*---*
+|   | X |   |   |
+*---*---*---*---*
+|   |   |   | X |
+*---*---*---*---*
+| X |   |   |   |
+*---*---*---*---*
+|   |   | X |   |
+*---*---*---*---*
+```
+
+## Algorithm
+
+1. Place the first queen at the first cell of the chessboard.
+2. Proceed to the next row. Move from the first column to the Nth column and see if that cell is safe or not so that no two queens are in an attacking position. If we find such a cell, then place the queen there and repeat step 2.
+3. If we cannot find any safe cell for that row, then go to the previous row, change the queen's position, and return false and repeat step 2.
+4. If we have successfully placed all N queens, then print the queens' corresponding positions and return true.
+5. If all the rows are tried and no way exists, then return false.
+
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+#define N 20
+
+void solve(bool board[N][N], bool cols[N], bool ndiag[2*N-1], bool rdiag[2*N-1], int row, string asf, int n) {
+
+    if(row == n) {
+        cout << asf << '.' << '\n';
+        return;
+    }
+
+    for(int col=0; col<n; col++) {
+        if(cols[col] == false && ndiag[row+col] == false &&
+            rdiag[row-col+n-1] == false) {
+                // place the queen
+                cols[col] = true;
+                ndiag[row+col] = true;
+                rdiag[row-col+n-1] = true;
+                board[row][col] = true;
+                solve(board, cols, ndiag, rdiag, row+1, asf+to_string(row)+"-"+to_string(col)+", ", n);
+                // backtrack
+                cols[col] = false;
+                ndiag[row+col] = false;
+                rdiag[row-col+n-1] = false;
+                board[row][col] = false;
+            }
+    }
+}
+
+int main() {
+	int n;
+	cin >> n;
+
+	bool board[N][N];
+    bool cols[N];
+    bool ndiag[2*N-1], rdiag[2*N-1];
+
+	// initialize
+	for(int i = 0; i < n; i++)
+		for(int j = 0; j < n; j++)
+			board[i][j] = false;
+
+    // initialize that no column has a queen
+	for(int i = 0; i < n; i++)
+		cols[i] = false;
+
+	for(int i = 0; i < 2*n-1; i++) {
+		ndiag[i] = false;
+		rdiag[i] = false;
+	}
+
+	solve(board, cols, ndiag, rdiag, 0, "", n);
```

Comment:
The input and output statements of C++ and Python don't match. Kindly fix them up